lib/matplotlib/ticker.py

```diff
@@ -1905,6 +1922,11 @@ def _staircase(steps):
 
     def set_params(self, **kwargs):
         """Set parameters within this locator."""
+        if set(kwargs) - self._supported_kwargs:
```

Comment:
You could just pop the kwargs at every if clause below and then check if kwargs is empty.  (Note that this check will need to stay even after the deprecation period is over and the signature of init changed, as we need to distinguish between a kwarg not passed to set_params and a kwarg set to its default.)